include/aikido/robot/ros/RosRobot.hpp

```diff
@@ -18,21 +21,198 @@ AIKIDO_DECLARE_POINTERS(RosRobot)
+      const std::string name)
+    : Robot(refSkeleton, rootRobot, collisionDetector, collisionFilter, name)
+  {
+  }
 
   /// Destructor.
   virtual ~RosRobot() = default;
+
+  ///
+  /// Steps the ros robot (and underlying executors and subrobots) through time.
+  /// Call regularly to update the state of the ros robot.
+  ///
+  /// \param[in] timepoint The point in time to step to.
+  void step(const std::chrono::system_clock::time_point& timepoint) override;
+
+  ///
+  /// Registers a subset of the joints of the skeleton as a new RosRobot.
+  /// Must be disjoint from other subrobots.
+  ///
+  /// \param[in] metaSkeleton The referential skeleton corresponding to the
+  /// subrobot. \param[in] name Name of the subrobot.
+  RobotPtr registerSubRobot(
+      dart::dynamics::ReferentialSkeletonPtr refSkeleton,
+      const std::string& name) override;
+
+  ///
+  /// Stops the controller corresponding to this executor and deactivates
+  /// executor.
+  void deactivateExecutor() override;
+
+  ///
+  /// Loads the controller. Throws a runtime_error if controller cannot be
+  /// loaded. Add an (executor, controller, control mode) to the inactive
+  /// (executor, controller, control mode) list Releases DoFs held by executor
+  /// if held.
+  ///
+  /// \param[in] executor The Executor to add to the inactive list.
+  /// \param[in] controllerName The corresponding controller to load and add to
+  /// the inactive list. \param[in] controllerMode The corresponding control
+  /// mode to add to the inactive list. \return A robot-unique non-negative ID
+  /// (negative implies failure)
+  std::string registerExecutor(
+      aikido::control::ExecutorPtr executor,
+      std::string controllerName,
+      hardware_interface::JointCommandModes controllerMode,
+      std::string desiredName);
+
+  ///
+  /// Loads the controller. Throws a runtime_error if controller cannot be
+  /// loaded. Add an (executor, controller) to the inactive (executor,
+  /// controller, control mode) list Operations on this executor do not affect
+  /// the control mode. Releases DoFs held by executor if held.
+  ///
+  /// \param[in] executor The Executor to add to the inactive list.
+  /// \param[in] controllerName The corresponding controller to load and add to
+  /// the inactive list. \return A robot-unique non-negative ID (negative
+  /// implies failure)
+  std::string registerExecutor(
+      aikido::control::ExecutorPtr executor, 
+      std::string controllerName,
+      std::string desiredName);
+
+  ///
+  /// Add an executor to the inactive executors list.
+  /// Operations on this executor do not affect the controller or control mode.
+  /// Releases DoFs held by executor if held.
+  ///
+  /// \param[in] executor The Executor to add to the inactive list.
+  /// \return A robot-unique non-negative ID (negative implies failure)
+  std::string registerExecutor(aikido::control::ExecutorPtr executor, std::string desiredName = "") override;
```

Comment:
Remove this. It should just call the parent class (no need to load a controller or register any controller names / modes)